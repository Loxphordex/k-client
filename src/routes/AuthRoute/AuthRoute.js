import React from 'react'
import ApiServices from '../../services/api-services'

export default class AuthRoute extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      password: '',
      error: null,
    }
  }

  updateUsername = (username) => {
    this.setState({ username })
  }

  updatePassword = (password) => {
    this.setState({ password })
  }

  handleError = (error) => {
    console.log(error)
  }

  clearError = () => {
    this.setState({ error: null })
  }

  clearCreds = () => {
    this.setState({
      username: '',
      password: '',
    })
  }

  handleSubmitUser = (event) => {
    event.preventDefault()

    const { username, password } = this.state
    const { history } = this.props

    ApiServices.userRegistration(username, password)
      .then(() => this.clearError())
      .then(() => this.clearCreds())
      .then(() => history.push('/'))
      .catch(error => this.handleError(error))
  }

  render() {
    const { error } = this.state
    return(
      <div>
        <h2>Auth</h2>
        { error && <div>{ error }</div> }
        <fieldset>
          <form onSubmit={(event) => this.handleSubmitUser(event)}>
            <input 
              type='text' 
              onChange={event => this.updateUsername(event.target.value)}
              placeholder='USERNAME'
              required />
            <input 
              type='password' 
              onChange={event => this.updatePassword(event.target.value)}
              placeholder='PASSWORD'
              required />
            <button type='submit'>SUBMIT</button>
          </form>
        </fieldset>
      </div>
    )
  }
}