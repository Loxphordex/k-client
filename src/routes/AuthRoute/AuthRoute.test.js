import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import AuthRoute from './AuthRoute'

configure({ adapter: new Adapter() })

describe('Gallery', () => {
  it('renders the complete page', () => {
    const wrapper = shallow(<AuthRoute />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<AuthRoute />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})
