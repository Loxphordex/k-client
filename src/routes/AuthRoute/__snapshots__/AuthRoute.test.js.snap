// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery renders the complete page 1`] = `
<div>
  <h2>
    Auth
  </h2>
  <fieldset>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="USERNAME"
        required={true}
        type="text"
      />
      <input
        onChange={[Function]}
        placeholder="PASSWORD"
        required={true}
        type="password"
      />
      <button
        type="submit"
      >
        SUBMIT
      </button>
    </form>
  </fieldset>
</div>
`;
